pkg/apis/workflow/v1alpha1/int64orstr.go

```diff
@@ -33,12 +33,12 @@ func (i *Int64OrString) UnmarshalJSON(value []byte) error {
 		*i = Int64OrString(v)
 		return nil
 	}
-	v := 0
+	v := 0.0
 	err := json.Unmarshal(value, &v)
 	if err != nil {
 		return err
 	}
-	*i = Int64OrString(strconv.Itoa(v))
+	*i = Int64OrString(strconv.FormatFloat(v, 'f', -1, 64))
```

Comment:
could we rename this type `AnyString` and add support for `bool`?

Reply:
Done